server/src/main/java/io/crate/expression/reference/doc/lucene/SourceLookup.java

```diff
@@ -94,6 +94,9 @@ private void ensureDocVisited() {
     }
 
     static Object extractValue(final Map<?, ?> map, List<String> path, int pathStartIndex) {
+        if (map == null) {
```

Comment:
In which case is the map null? Isn't the change in the SourceParser enough?

Reply:
It receives what I returned from parse, so it's null when bytes is null  - and without that check NPE still happens on `m.get(path.get(i));`